src/psyclone/psyGen.py

```diff
@@ -1924,11 +1924,30 @@ def gen_code(self, parent):
                 module_map[module_name].append(globalvar.name)
             else:
                 module_map[module_name] = [globalvar.name]
+
         # Then we can produce the UseGen statements without repeating modules
         for module_name, var_list in module_map.items():
             parent.add(UseGen(parent, name=module_name, only=True,
                               funcnames=var_list))
 
+        # Add the SymbolTable symbols used into the NameSpaceManager to prevent
+        # clashes. Note that the global variables names are going to be used as
+        # arguments and are declared with 'AlgArgs' context.
+        # TODO: After #312 this will not be necessary.
+        for module_name, var_list in module_map.items():
+            self._name_space_manager.add_reserved_name(module_name)
+            for var_name in var_list:
+                newname = self._name_space_manager.create_name(
+                    root_name=var_name,
+                    context="AlgArgs",
+                    label=var_name)
+                # There is a name clash with this variable name and we can not
+                # accept indexed names for global variables.
```

Comment:
Please add a TODO (and new issue if necessary) here as we should try to work around this. e.g. in Fortran you can re-name the variable being imported with a USE. However, that wouldn't help us if the module name itself clashed.

Reply:
Added TODO and created issue #642